### DISCUSSION TITLE
Adapt name of field put into MDC for trace_id

### DISCUSSION THREAD

**Discussion by huehnerlady** [open-telemetry/opentelemetry-java-instrumentation#8402] [Q&A] [answered]
Hi,

We are now using the javaagent and I can see that the trace-id is put into MDC under the label `trace_id`.

Unfortunately we still have an old trace-id, which also runs under that same label.
I was wondering if there is a chance to configure that label using a property? I couldn't find anything in the parameters list.
But maybe there is a trick?

**Comment by laurit** [comment, marked as answer]
Currently there is no way to customize the MDC keys for `trace_id` etc. There is https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/7985 but it is unknown when or whether this will be accepted. I think that this really isn't a big issue. For modern versions of both log4j and logback adding stuff to MDC is not hard, you can easily copy paste our instrumentations and modify them as you see fit and disable what the agent provides. See https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/instrumentation/log4j/log4j-context-data/log4j-context-data-2.17/library-autoconfigure and https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/instrumentation/logback/logback-mdc-1.0/library

**Reply by huehnerlady** [reply]
Hi thanks for the answer, 

the problem is that we have a huge amount of services and want to keep it as simple as possible for now - hence the javaagent approach instead of dependencies. So I guess we see if we can change the name of our trace_id. 
Or we go your suggested way after all 👍🏻
